File: bench/fib.lua
-- The Lua twin of fib.sl. `local function` rather than `local fib = function` so the name is in
-- scope inside its own body, which is what makes the recursion work in Lua.

local function fib(n)
    if n < 2 then
        return n
    end

    return fib(n - 1) + fib(n - 2)
end

print(fib(33))

File: bench/fib.php
<?php
// The PHP twin of fib.sl. A PHP call does not recurse on the C stack, and 33 deep is nothing to its
// own, so what this measures is the cost of a frame.

function fib($n)
{
    return $n < 2 ? $n : fib($n - 1) + fib($n - 2);
}

echo fib(33), "\n";
